mypy/main.py

```diff
@@ -351,125 +354,249 @@ def add_invertible_flag(flag: str,
+             'type parameters')
+
+    untyped_group = parser.add_argument_group(
+        title='untyped definitions and calls',
+        description="Configure how untyped definitions and calls are handled.")
     add_invertible_flag('--disallow-untyped-calls', default=False, strict_flag=True,
                         help="disallow calling functions without type annotations"
-                        " from functions with type annotations")
+                        " from functions with type annotations",
+                        group=untyped_group)
     add_invertible_flag('--disallow-untyped-defs', default=False, strict_flag=True,
                         help="disallow defining functions without type annotations"
-                        " or with incomplete type annotations")
+                        " or with incomplete type annotations",
+                        group=untyped_group)
     add_invertible_flag('--disallow-incomplete-defs', default=False, strict_flag=True,
-                        help="disallow defining functions with incomplete type annotations")
+                        help="disallow defining functions with incomplete type annotations",
+                        group=untyped_group)
     add_invertible_flag('--check-untyped-defs', default=False, strict_flag=True,
-                        help="type check the interior of functions without type annotations")
-    add_invertible_flag('--disallow-subclassing-any', default=False, strict_flag=True,
-                        help="disallow subclassing values of type 'Any' when defining classes")
+                        help="type check the interior of functions without type annotations",
+                        group=untyped_group)
     add_invertible_flag('--warn-incomplete-stub', default=False,
                         help="warn if missing type annotation in typeshed, only relevant with"
-                        " --check-untyped-defs enabled")
-    add_invertible_flag('--disallow-untyped-decorators', default=False, strict_flag=True,
-                        help="disallow decorating typed functions with untyped decorators")
+                             " --check-untyped-defs enabled",
+                        group=untyped_group)
+
+    none_group = parser.add_argument_group(
+        title='handle None and Optional',
+        description="Adjust how values of type 'None' are handled.")
+    add_invertible_flag('--no-implicit-optional', default=False, strict_flag=True,
+                        help="don't assume arguments with default values of None are Optional",
+                        group=none_group)
+    none_group.add_argument(
+        '--strict-optional', action='store_true',
+        help=argparse.SUPPRESS)
+    none_group.add_argument(
+        '--no-strict-optional', action='store_false', dest='strict_optional',
+        help="disable strict Optional checks (inverse: --strict-optional)")
+    none_group.add_argument(
```

Comment:
We should kill this relic of not having per-module configuration

Reply:
We should probably deprecate it for at least a release.

Reply:
[meta] When you comment on a multi-line thing please put your comment on the last line -- the GitHub email and Conversation tab only show code above the line you commented on.